pkg/virt-launcher/virtwrap/util/libvirt_helper.go

```diff
@@ -104,10 +104,27 @@ func SetDomainSpec(virConn cli.Connection, vmi *v1.VirtualMachineInstance, wante
 	return dom, nil
 }
 
+// GetDomainSpecWithRuntimeInfo return the active domain XML with runtime information embedded
+func GetDomainSpecWithRuntimeInfo(status libvirt.DomainState, dom cli.VirDomain) (*api.DomainSpec, error) {
+
+	// get libvirt xml with runtime status
+	activeSpec, err := GetDomainSpecWithFlags(dom, 0)
```

Comment:
As far as I know, and as David said, this will not contain the up-to-date metadata from libvirt. Only the state of the metadata it used to have when the domain started. We dynamically update the metadata, so virt-handler needs to see the up-to-date information.

Reply:
Yes, I was looking to the vdsm for an answer and found it. The function is now updated with GetMetadata call, which work everytime and does not have to go through building the whole XML.